geopy/geocoders/geocodefarm.py

```diff
@@ -159,6 +161,7 @@ def _check_for_api_errors(geocoding_results):
         in the api response.
         """
         status_result = geocoding_results.get("STATUS", {})
+        if "NO_RESULTS" in status_result.get("status", ""): return
```

Comment:
Is there any reason to have this check here? If there's none, I suggest to remove this line.

As far as I can see the following two lines shouldn't misbehave even when the status contains `NO_RESULTS`.

Reply:
Sorry, if I'm being honest this was long enough ago that I forgot my rational for the double lines...  I'll figure that out and cleanup.

Reply:
@KostyaEsmukov I double checked, both checks are necessary.